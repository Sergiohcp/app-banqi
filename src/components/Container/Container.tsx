import React from "react";
import { View, FlexAlignType, StyleProp, ViewStyle } from "react-native";

interface IContainer {
  flex?: number;
  flexDirection?: "row" | "column" | "row-reverse" | "column-reverse";
  justifyContent?:
    | "flex-start"
    | "flex-end"
    | "center"
    | "space-between"
    | "space-around"
    | "space-evenly";
  alignItems?: FlexAlignType;
  mt?: number;
  ml?: number;
  mr?: number;
  mb?: number;
  p?: number;
  borderRadius?: number;
  style?: StyleProp<ViewStyle>;
  shadow?: boolean;
  bgColor?: string;
}

const Container: React.FC<IContainer> = ({ children, ...rest }) => {
  const {
    flex = 0,
    justifyContent = "flex-start",
    alignItems = "flex-start",
    mt = 0,
    ml = 0,
    mr = 0,
    mb = 0,
    p = 0,
    borderRadius = 0,
    style,
    shadow = false,
    bgColor = "transparent",
  } = rest;

  let viewStyle: ViewStyle = {
    flex,
    justifyContent: justifyContent,
    alignItems,
    marginTop: mt,
    marginLeft: ml,
    marginRight: mr,
    marginBottom: mb,
    padding: p,
    borderRadius,
    backgroundColor: bgColor,
  };

  if (shadow) {
    viewStyle = {
      ...viewStyle,
      shadowColor: "#000000",
      shadowOffset: {
        width: 0,
        height: 2,
      },
      shadowOpacity: 0.23,
      shadowRadius: 2.62,
      elevation: 4,
    };
  }

  return <View style={[viewStyle, style]}>{children}</View>;
};

export default Container;
