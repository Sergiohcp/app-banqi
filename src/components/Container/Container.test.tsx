import React from "react";
import renderer from "react-test-renderer";

import Container from "./Container";

describe("Container test", () => {
  it("should take snapshot of component", () => {
    const tree = renderer
      .create(<Container width={25} height={25} bgColor={"pink"} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
