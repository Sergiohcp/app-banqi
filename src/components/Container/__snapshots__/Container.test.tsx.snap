// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container test should take snapshot of component 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": undefined,
        "alignSelf": "auto",
        "backgroundColor": "#EC008C",
        "borderRadius": undefined,
        "flex": 0,
        "flexDirection": "column",
        "height": 25,
        "justifyContent": "flex-start",
        "marginBottom": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
        "width": 25,
      },
      undefined,
    ]
  }
/>
`;
